aspnetcore-authentication-jwtbearer/src/AspNetCore.Authentication.JwtBearer/DPoP/DPoPServiceCollectionExtensions.cs

```diff
@@ -13,40 +14,63 @@ namespace Duende.AspNetCore.Authentication.JwtBearer.DPoP;
 /// </summary>
 public static class DPoPServiceCollectionExtensions
 {
-    /// <summary>
-    /// Sets up DPoP on a JwtBearer authentication scheme.
-    /// </summary>
-    public static IServiceCollection ConfigureDPoPTokensForScheme(this IServiceCollection services, string scheme)
+    extension(IServiceCollection services)
     {
-        services.AddOptions<DPoPOptions>();
-
-        services.AddTransient<DPoPJwtBearerEvents>();
-        services.TryAddTransient<IDPoPProofValidator, DPoPProofValidator>();
-        services.TryAddTransient<IDPoPNonceValidator, DefaultDPoPNonceValidator>();
-        services.AddTransient<DPoPExpirationValidator>();
-        services.TryAddTransient<IDPoPProofValidator, DPoPProofValidator>();
-        services.AddDistributedMemoryCache();
-        services.AddHybridCache();
-        services.TryAddTransient<IReplayCache, ReplayCache>();
-
-        services.AddSingleton<ConfigureJwtBearerOptions>();
-        services.AddSingleton<IPostConfigureOptions<JwtBearerOptions>>(sp =>
+        /// <summary>
+        /// Sets up DPoP on a JwtBearer authentication scheme.
+        /// </summary>
+        public IServiceCollection ConfigureDPoPTokensForScheme(string scheme)
         {
-            var opt = sp.GetRequiredService<ConfigureJwtBearerOptions>();
-            opt.Scheme = scheme;
-            return opt;
-        });
-        services.AddSingleton<IPostConfigureOptions<JwtBearerOptions>, ConfigureJwtBearerOptions>();
+            services.AddOptions<DPoPOptions>();
 
-        return services;
-    }
+            services.AddTransient<DPoPJwtBearerEvents>();
+            services.TryAddTransient<IDPoPProofValidator, DPoPProofValidator>();
+            services.TryAddTransient<IDPoPNonceValidator, DefaultDPoPNonceValidator>();
+            services.AddTransient<DPoPExpirationValidator>();
+            services.TryAddTransient<IDPoPProofValidator, DPoPProofValidator>();
 
-    /// <summary>
-    /// Sets up DPoP on a JwtBearer authentication scheme, and configures <see cref="DPoPOptions"/>.
-    /// </summary>
-    public static IServiceCollection ConfigureDPoPTokensForScheme(this IServiceCollection services, string scheme, Action<DPoPOptions> configure)
-    {
-        services.Configure(scheme, configure);
-        return services.ConfigureDPoPTokensForScheme(scheme);
+            services.AddKeyedHybridCache(
+                serviceKey: ServiceProviderKeys.ProofTokenReplayHybridCache,
+                opt => opt.DistributedCacheServiceKey = ServiceProviderKeys.ProofTokenReplayDistributedCache);
+
+            services.TryAddTransient<IReplayCache, ReplayCache>();
+
+            services.AddSingleton<ConfigureJwtBearerOptions>();
+
+            services.AddSingleton<IPostConfigureOptions<JwtBearerOptions>>(sp =>
+            {
+                var distributedCache =
+                    sp.GetKeyedService<IDistributedCache>(ServiceProviderKeys.ProofTokenReplayDistributedCache);
```

Comment:
Why does JwtBearer support both IDistributedCache AND HybridCache? And why does it look like the IDistributedCache is required?